packages/internal/src/build/babel/common.ts

```diff
@@ -30,3 +32,47 @@ interface BabelRegisterOptions extends TransformOptions {
 export const registerBabel = (options: BabelRegisterOptions) => {
   require('@babel/register')(options)
 }
+
+export const CORE_JS_VERSION = pkgJson.dependencies['core-js']
+  .split('.')
+  .slice(0, 2)
+  .join('.') // Produces: 3.12, instead of 3.12.1
+
+if (!CORE_JS_VERSION) {
+  throw new Error(
+    'RedwoodJS Project Babel: Could not determine core-js version.'
+  )
+}
+
+const RUNTIME_CORE_JS_VERSION = pkgJson.dependencies['@babel/runtime-corejs3']
+if (!RUNTIME_CORE_JS_VERSION) {
+  throw new Error(
+    'RedwoodJS Project Babel: Could not determine core-js runtime version'
+  )
+}
+
+export const getCommonPlugins = () => {
+  return [
+    ['@babel/plugin-proposal-class-properties', { loose: true }],
+    // Note: The private method loose mode configuration setting must be the
+    // same as @babel/plugin-proposal class-properties.
+    // (https://babeljs.io/docs/en/babel-plugin-proposal-private-methods#loose)
+    ['@babel/plugin-proposal-private-methods', { loose: true }],
+    ['@babel/plugin-proposal-private-property-in-object', { loose: true }],
+
+    // Not sure about this one:
+    // Do we need this?
+    // [
+    //   '@babel/plugin-transform-runtime',
+    //   {
+    //     // https://babeljs.io/docs/en/babel-plugin-transform-runtime/#core-js-aliasing
+    //     // Setting the version here also requires `@babel/runtime-corejs3`
+    //     corejs: { version: 3, proposals: true },
+    //     // https://babeljs.io/docs/en/babel-plugin-transform-runtime/#version
+    //     // Transform-runtime assumes that @babel/runtime@7.0.0 is installed.
+    //     // Specifying the version can result in a smaller bundle size.
+    //     version: RUNTIME_CORE_JS_VERSION,
+    //   },
+    // ],
```

Comment:
🤷 I guess a test would be really helpful here to accurately describe what this does.

Reply:
I'm of the opposite opinion - I think it's better _not_ to include this, **unless** we're also including Babel/runtime as a dependency on the rwjs/api. These two things must go hand in hand - as we learnt from one our releases a while ago. 

I'm pretty sure the fact that this was still in the root Babel.config was wrong - and ignored for builds anyway.

Reply:
Yep as I mentioned @thedavidprice this needs `@babel/runtime` dep if we use this plugin. 

We can evaluate later on whether we should include this, but to maintain the same configuration as we've had past couple of releases, we should _not_ include this

Reply:
Ah, a lot going on here. Also, I didn't correctly understand the relationship with `@babel/runtime`

> I'm pretty sure the fact that this was still in the root Babel.config was wrong - and ignored for builds anyway. So adding it here would be a change 😂, confusing I know!

^^ Got it. And definitely an important consideration!

Happy to discuss real-time later this morning (Pacific time)! My primary concern is keeping babel config consistent with this PR. And now I better understand why you were making this (apparent) change.

Reply:
OK Update on this, thanks to @dthyresson for rubber ducking with me. We have 3 options here, either:

## Option A: remove polyfill plugin
We remove the polyfill plugin `babel-plugin-polyfill-corejs3` and include `babel-plugin-transform-runtime`. We will **also** need to change our _framework_ config to include `babel-plugin-transform-runtime`, because otherwise it won't use the correct module

We then only have @babel/runtime-corejs3 as a dependency on the api package.json.

**Why?**
- It seems the polyfill-corejs3 doesn't cover all cases of polyfilling. e.g. `Array.at()`
- polyfill plugin will add imports, where it isn't required (e.g. try doing `Promise.any`)
- We want to minimise the number of these polyfill dependencies, so the framework config must match the project config

## Option B: accept polyfill-corejs3 misvehaviour 
- We accept that this plugin will sometimes misbehave
- We include both `corejs` and `@babel/runtime-corejs3` as dependencies 
- Trade off is that we are adding both these dependencies at runtime (worth about 1.2MB) in the api side. The web side should be fine since we're bundling anyway

## Option C: not include transform-runtime plugin
- This is effectively what we have right now, but only have the polyfill-corejs3 plugin
- Accept that some things won't be polyfilled e.g. `Array.at()`

---

I don't think all of this really affects the web side all that much - since the webside uses preset-env anyway

Reply:
If I recall correctly (there's a 50/50 chance), the `babel-plugin-polyfill-corejs3` was added to correct for ESBuild behavior.

y/n/maybe?

Reply:
y

Reply:
Pushing up option A at the moment, until we discuss.